lib/compiler/objcopy.zig

```diff
@@ -1179,10 +1178,13 @@ fn ElfFile(comptime is_64: bool) type {
                                     const data = try allocator.alignedAlloc(u8, section_memory_align, src_data.len);
                                     @memcpy(data, src_data);
 
-                                    const defs = @as([*]Elf_Verdef, @ptrCast(data))[0 .. @as(usize, @intCast(src.sh_size)) / @sizeOf(Elf_Verdef)];
+                                    const defs = @as([*]elf.Verdef, @ptrCast(data))[0 .. @as(usize, @intCast(src.sh_size)) / @sizeOf(elf.Verdef)];
                                     for (defs) |*def| {
-                                        if (def.vd_ndx != elf.SHN_UNDEF)
-                                            def.vd_ndx = sections_update[src.sh_info].remap_idx;
+                                        // Original author of this next line had elf.SHN_UNDEF
+                                        // here which does not make sense given that this field
+                                        // is elf.VER_NDX
+                                        if (def.ndx != .LOCAL)
+                                            def.ndx = @enumFromInt(sections_update[src.sh_info].remap_idx);
```

Comment:
@xxxbxxx can you shed some light on this? does the VER_NDX type that I added to std.elf not make sense, or is using `elf.SHN_UNDEF` here incorrect?

Reply:
yes, indeed, the type of the '0' value is wrong. yeah for strong typing!

And it is actually the test is incomplete I think: '1' is also a special value not to be re-indexed,
According to:
 https://refspecs.linuxfoundation.org/LSB_3.1.1/LSB-Core-generic/LSB-Core-generic/symversion.html 
 

> 11.7.2. Symbol Version Table
> [...] These values are identifiers that are provided by [...] the `vd_ndx` member of the 
> `Elfxx_Verdef` structure.
> The values 0 and 1 are reserved.
> 
> 0 |   | The symbol is local, not available outside the object.
> -- | -- | --
> 1 |   | The symbol is defined in this object and is globally available.
> 
> 

-> so I guess the test should be `def.ndx != .LOCAL and def.ndx != .GLOBAL` .

Reply:
Thanks!